File: 7-ReverseInteger/7-ReverseInteger.py
# Last updated: 8/24/2025, 12:32:40 PM
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x < 0 else 1
        temp = abs(x)
        output = 0
        while temp > 0:
            output = (output * 10) + (temp % 10)
            temp = temp // 10
         
        if output < -2**31 or output > 2**31 - 1:
            return 0
        return sign * output

File: 7-ReverseInteger/7-ReverseInteger.java
// Last updated: 8/24/2025, 12:32:41 PM
class Solution {
    public int reverse(int x) { //123 --> 321
        
        long rev = 0;
        while(x!=0)
        {
                int digit = x % 10;
                rev = rev*10 + digit;
                x = x/10;
        }
        
        if(rev > (Math.pow(2, 31)-1) || rev < -(Math.pow(2, 31)) )
        {
            return 0;
        }
        
        return (int)rev;
     }
}